### PR TITLE
Support for multiple Maven versions

### PR DESCRIPTION
**Pull request by ReneSchumacher**
Description: The current Maven task does not support multiple installations of Maven on a single machine. The old TFS build extensions did support this. Adding that support to the new system will simplify the move from the old to the new build system.

Solution: Changes have been made to the JavaScript and PowerShell implementations of the task to support setting M2_HOME and run maven from the correct path. Additional config options have been added to the advanced group of the task, similar to the ones for the JDK. A German translation for the new options has been added, too.

Testing: Manual.

### CODE REVIEW

File: Tasks/Maven/Strings/resources.resjson/de-de/resources.resjson [hunk 1]
```diff
@@ -20,5 +20,11 @@
   "loc.input.label.jdkUserInputPath": "JDK Path",
   "loc.input.help.jdkUserInputPath": "Sets JAVA_HOME to the given path.",
   "loc.input.label.jdkArchitecture": "JDK-Architektur",
-  "loc.input.help.jdkArchitecture": "Geben Sie optional die JDK-Architektur an (x86, x64)."
+  "loc.input.help.jdkArchitecture": "Geben Sie optional die JDK-Architektur an (x86, x64).",
```
Comment (bryanmacfarlane) [inline Tasks/Maven/Strings/resources.resjson/de-de/resources.resjson]: typically we don't check non-English strings in.  They generated English strings file get's handed off to loc and then they check back in.

**Comment by bryanmacfarlane** [comment]
I'll defer to folks on the java team.  Assigning to Yang to redirect, test and approve

File: Tasks/Maven/maven.ps1 [hunk 2]
```diff
@@ -13,6 +13,9 @@
     [string]$jdkVersion,
     [string]$jdkArchitecture,
     [string]$jdkUserInputPath, 
+	[string]$mavenVersionSelection,
+	[string]$mavenPath,
+	[string]$mavenSetM2Home,
```
Comment (jeffyoung) [inline Tasks/Maven/maven.ps1]: It'd be great to fix up the whitespace here.

**Comment by jeffyoung** [comment]
Hi Rene.  You'll want to increase the version of the task in the task.json and task.loc.json file so that the updated task will get down to the agents.  I don't recall if the task.loc.json is updated by the build automatically or not.  I'm going to build your changes locally and do a bit of testing and I'll report back.  Thanks for issuing the request!

File: Tasks/Maven/Strings/resources.resjson/de-de/resources.resjson (hunk 1 above)
Comment (ReneSchumacher) [inline Tasks/Maven/Strings/resources.resjson/de-de/resources.resjson]: Should I remove those changes from the branch (e.g., rebase) then?

File: Tasks/Maven/maven.ps1 (hunk 2 above)
Comment (ReneSchumacher) [inline Tasks/Maven/maven.ps1]: Will do.

File: Tasks/Maven/Strings/resources.resjson/de-de/resources.resjson (hunk 1 above)
Comment (jeffyoung) [inline Tasks/Maven/Strings/resources.resjson/de-de/resources.resjson]: Yeah, go ahead and do that.  Thanks Rene!

**Comment by ReneSchumacher** [comment]
Hi Jeff! I updated the version and removed my changes from the localized resource file.

**Pull request merged**